.travis.yml

```diff
@@ -23,7 +23,11 @@ before_script:
   - phpenv config-rm xdebug.ini
 
 script:
-  - if [[ "$VALIDATION" == "true" ]]; then phpunit --testsuite validation; fi
-  - if [[ "$VALIDATION" != "true" ]]; then phpunit --testsuite invariants; fi
-  - if [[ "$VALIDATION" != "true" ]]; then phpunit --testsuite grammar; fi
-  - if [[ "$VALIDATION" != "true" ]]; then phpunit --testsuite api; fi
+  - |
+    if [[ $VALIDATION = true ]]; then
+      phpunit --testsuite validation
+    else
+      phpunit --testsuite invariants
+      phpunit --testsuite grammar
+      phpunit --testsuite api
+    fi
```

Comment:
I suppose that's true: the `if` statement returns the exit code of the last command only. Some people chain commands together with `&&` to avoid this, but I think `set -e` is a cleaner approach. I believe Travis compiles every step into one big bash script, in which case you may prefer to specify `set -e` at the earliest stage rather than just the `script` step.